compiler/src/iree/compiler/PluginAPI/Client.h

```diff
@@ -58,6 +62,13 @@ class PipelineExtensions {
   // considered valid.
   virtual void populateCustomInputConversionTypes(StringSet<> &typeMnemonics) {}
 
+  // Returns an appropriate typeMnemonic if operations from a dialect this
+  // instance supports are present in the module.
+  // Requires that dialects have already been added using |registerDialects|.
+  virtual std::string detectCustomInputOperations(ModuleOp &module) {
```

Comment:
I think most here are fine as we can relatively freely change, this one is something that spans across repos.

I don't like "custom" here, I don't know what else to call. Reading the description, it feels like "is supported" or "is required" ... but yes still tangled to do this. It almost feels like if a plugin says it doesn't support then it doesn't get added, else it is used to populate pipeline. Currently we'd have only 1 say required (e.g., only if stablehlo then stablehlo would say required) on a supported path. Now if input is mixed then it won't work but ideally all would be partial conversions and these could be composed.

Reply:
I was trying to match the other API methods, and two of them use "custom" already:

```c++
virtual void registerDialects(DialectRegistry &registry);
virtual void extendInputConversionPreprocessingPassPipeline(
    OpPassManager &passManager, InputDialectOptions::Type inputType);
virtual void populateCustomInputConversionTypes(StringSet<> &typeMnemonics);

// New:
virtual std::string detectCustomInputOperations(ModuleOp &module);

virtual bool
    extendCustomInputConversionPassPipeline(OpPassManager &passManager,
                                            std::string_view typeMnemonic);
virtual void extendPreprocessingPassPipeline(OpPassManager &passManager);
```

Good points though. How about `detectCustomInputConversionTypes` that returns all of the "custom input conversion types" from `populateCustomInputConversionTypes` that are detected within a module? Any type that function returns could then be used with `extendCustomInputConversionPassPipeline`.

```c++
virtual StringSet<> detectCustomInputConversionTypes(ModuleOp &module);
```

(could return a different data structure)